core/database_util.go

```diff
@@ -164,6 +95,20 @@ func GetHeadFastBlockHash(db ethdb.Database) common.Hash {
 	return common.BytesToHash(data)
 }
 
+// GetHeadBlockNum retrieves the block number of the current canonical head block.
+func GetHeadBlockNum(db ethdb.Database) *big.Int {
```

Comment:
Actually to add to that, I don't really understand what this method does at all:

 * The `headBlockKey` database entry maps "LastBlock" to the hash of the block (https://github.com/ethereum/go-ethereum/blob/develop/core/database_util.go#L147)
 * Even if the key would map to a block, why would we try and decode it as a header?